examples/python/images/image_writer.py

```diff
@@ -0,0 +1,64 @@
+###############################################################################
+# (c) 2019 Copyright, Real-Time Innovations.  All rights reserved.       #
+# No duplications, whole or partial, manual or electronic, may be made        #
+# without express written permission.  Any such copies, or revisions thereof, #
+# must display this notice unaltered.                                         #
+# This code contains trade secrets of Real-Time Innovations, Inc.             #
+###############################################################################
+
+"""Simple Square writer"""
+
+from sys import path as sys_path
+from os import path as os_path
+from os import getpid
+from time import sleep
+import numpy as np, random
+
+# Updating the system path is not required if you have pip-installed
+# rticonnextdds-connector
+file_path = os_path.dirname(os_path.realpath(__file__))
+sys_path.append(file_path + "/../../../")
+import rticonnextdds_connector as rti
+
+def random_rgb():
+    # Generate a random RGB pixel
+    return [int(255*random.random()), int(255*random.random()), int(255*random.random())]
+
+def generate_random_image(size):
+    # Generate a random flat list of pixels of 3 different colors
+    random_colors = [random_rgb() for i in range(3)]
+    pixels = np.array([random.choice(random_colors) for i in range(size)])
+    return pixels.reshape(-1).tolist() # Flatten out and convert to list
+
+def update_image(pixels, iteration):
+    # Shift 3 elements (one pixel)
+    return pixels[3:] + pixels[:3]
+
+with rti.open_connector(
+    config_name = "MyParticipantLibrary::ImagePubParticipant",
+    url = file_path + "/ImagesExample.xml") as connector:
+
+    output = connector.get_output("MyPublisher::ImageWriter")
+
```

Comment:
wait for discovery?

Reply:
It's not necessary. I didn't want to give the impression that you always need to wait for discovery. It could give the impression to beginners that DDS is point-to-point or brokered instead of completely asynchronous peer-to-peer.

Reply:
In fact I think in this case not waiting for discovery is more realistic. This is a best-effort "streaming" source of data. So the application sends the "live" images as soon as you run the app; it doesn't start producing them when someone starts listening.